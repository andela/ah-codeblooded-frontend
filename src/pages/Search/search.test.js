import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';

import { Search } from './index';
import mockStore from '../../utils/redux_mock_store';

const store = mockStore();

const props = {
  fetchArticles: jest.fn(),
};

describe('the search container', () => {
  const wrapper = mount(<Provider store={store}><Search {...props} /></Provider>);

  beforeEach(() => {
    wrapper.setProps({ ...props });
  });

  it('renders without crashing', () => {
    const wrapper2 = shallow(<Search {...props} />);
    expect(wrapper2).toMatchSnapshot();
  });

  it('receive props', () => {
    expect(wrapper.props()).toBeTruthy();
  });

  it('always renders a div', () => {
    const divs = wrapper.find('div');
    expect(divs).toBeDefined();
  });

  it('renders a Navbar component', () => {
    const NavBar = wrapper.find('Navbar');
    expect(NavBar).toBeTruthy();
  });

  it('renders a ConnectedArticleListing component', () => {
    const ArticleListing = wrapper.find('ArticleListing');
    expect(ArticleListing).toBeTruthy();
  });

  it('renders a search icon', () => {
    const icon = wrapper.find('i').text();
    expect(icon).toBe('search');
  });

  it('calls onSearch method', () => {
    wrapper.find('input').simulate('change');
    expect(props.fetchArticles).toHaveBeenCalled();
  });
});
