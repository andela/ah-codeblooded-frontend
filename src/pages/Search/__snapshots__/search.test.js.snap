// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the search container renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    fetchArticles={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(NavBar) />,
        <div
          className="container"
        >
          <div
            className="nav-wrapper"
          >
            <form>
              <div
                className="input-field"
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here"
                  required={true}
                  type="search"
                  value=""
                />
                <label
                  className="label-icon"
                  htmlFor="search"
                >
                  <i
                    className="material-icons"
                  >
                    search
                  </i>
                </label>
              </div>
            </form>
          </div>
        </div>,
        <div
          className="container"
        >
          <div
            className="row"
          >
            <Connect(ArticleListing)
              emptyMessage="We did not find any results"
              initialList={5}
              listName="search"
              url="articles/search_filter"
            />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nav-wrapper"
          >
            <form>
              <div
                className="input-field"
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here"
                  required={true}
                  type="search"
                  value=""
                />
                <label
                  className="label-icon"
                  htmlFor="search"
                >
                  <i
                    className="material-icons"
                  >
                    search
                  </i>
                </label>
              </div>
            </form>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              <div
                className="input-field"
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here"
                  required={true}
                  type="search"
                  value=""
                />
                <label
                  className="label-icon"
                  htmlFor="search"
                >
                  <i
                    className="material-icons"
                  >
                    search
                  </i>
                </label>
              </div>
            </form>,
            "className": "nav-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="input-field"
              >
                <input
                  name="search"
                  onChange={[Function]}
                  placeholder="Search here"
                  required={true}
                  type="search"
                  value=""
                />
                <label
                  className="label-icon"
                  htmlFor="search"
                >
                  <i
                    className="material-icons"
                  >
                    search
                  </i>
                </label>
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here"
                    required={true}
                    type="search"
                    value=""
                  />,
                  <label
                    className="label-icon"
                    htmlFor="search"
                  >
                    <i
                      className="material-icons"
                    >
                      search
                    </i>
                  </label>,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "search",
                    "onChange": [Function],
                    "placeholder": "Search here",
                    "required": true,
                    "type": "search",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="material-icons"
                    >
                      search
                    </i>,
                    "className": "label-icon",
                    "htmlFor": "search",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "search",
                      "className": "material-icons",
                    },
                    "ref": null,
                    "rendered": "search",
                    "type": "i",
                  },
                  "type": "label",
                },
              ],
              "type": "div",
            },
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <Connect(ArticleListing)
              emptyMessage="We did not find any results"
              initialList={5}
              listName="search"
              url="articles/search_filter"
            />
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(ArticleListing)
              emptyMessage="We did not find any results"
              initialList={5}
              listName="search"
              url="articles/search_filter"
            />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "emptyMessage": "We did not find any results",
              "initialList": 5,
              "listName": "search",
              "url": "articles/search_filter",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(NavBar) />,
          <div
            className="container"
          >
            <div
              className="nav-wrapper"
            >
              <form>
                <div
                  className="input-field"
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here"
                    required={true}
                    type="search"
                    value=""
                  />
                  <label
                    className="label-icon"
                    htmlFor="search"
                  >
                    <i
                      className="material-icons"
                    >
                      search
                    </i>
                  </label>
                </div>
              </form>
            </div>
          </div>,
          <div
            className="container"
          >
            <div
              className="row"
            >
              <Connect(ArticleListing)
                emptyMessage="We did not find any results"
                initialList={5}
                listName="search"
                url="articles/search_filter"
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="nav-wrapper"
            >
              <form>
                <div
                  className="input-field"
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here"
                    required={true}
                    type="search"
                    value=""
                  />
                  <label
                    className="label-icon"
                    htmlFor="search"
                  >
                    <i
                      className="material-icons"
                    >
                      search
                    </i>
                  </label>
                </div>
              </form>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form>
                <div
                  className="input-field"
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here"
                    required={true}
                    type="search"
                    value=""
                  />
                  <label
                    className="label-icon"
                    htmlFor="search"
                  >
                    <i
                      className="material-icons"
                    >
                      search
                    </i>
                  </label>
                </div>
              </form>,
              "className": "nav-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="input-field"
                >
                  <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search here"
                    required={true}
                    type="search"
                    value=""
                  />
                  <label
                    className="label-icon"
                    htmlFor="search"
                  >
                    <i
                      className="material-icons"
                    >
                      search
                    </i>
                  </label>
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      name="search"
                      onChange={[Function]}
                      placeholder="Search here"
                      required={true}
                      type="search"
                      value=""
                    />,
                    <label
                      className="label-icon"
                      htmlFor="search"
                    >
                      <i
                        className="material-icons"
                      >
                        search
                      </i>
                    </label>,
                  ],
                  "className": "input-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "search",
                      "onChange": [Function],
                      "placeholder": "Search here",
                      "required": true,
                      "type": "search",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        className="material-icons"
                      >
                        search
                      </i>,
                      "className": "label-icon",
                      "htmlFor": "search",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "search",
                        "className": "material-icons",
                      },
                      "ref": null,
                      "rendered": "search",
                      "type": "i",
                    },
                    "type": "label",
                  },
                ],
                "type": "div",
              },
              "type": "form",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <Connect(ArticleListing)
                emptyMessage="We did not find any results"
                initialList={5}
                listName="search"
                url="articles/search_filter"
              />
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(ArticleListing)
                emptyMessage="We did not find any results"
                initialList={5}
                listName="search"
                url="articles/search_filter"
              />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "emptyMessage": "We did not find any results",
                "initialList": 5,
                "listName": "search",
                "url": "articles/search_filter",
              },
              "ref": [Function],
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
